src/ui/Templates/TemplateFromAScriptTag.ts

```diff
@@ -35,6 +36,8 @@ export class TemplateFromAScriptTag {
     this.template.desktop = this.parseScreenSize('data-desktop');
     this.template.fields = TemplateConditionEvaluator.getFieldFromString(`${scriptTag.innerHTML} ${condition ? condition : ''}`);
 
+    this.template.role = <TemplateRole>scriptTag.getAttribute('data-role');
```

Comment:
What happen if the `data-role` does not match `template-footer` or `template-header` (since it is reading an attribute set arbitrarily on a script tag).